src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -1046,12 +1048,25 @@ public static SyntaxKind GetPreprocessorKeywordKind(string text)
             }
         }
 
+        private static ReadOnlyCollection<SyntaxKind> _lazyContextualKeywordKindsArray;
```

Comment:
You can just have this be an IReadOnlyList or IEnumerable.  But just initialize it with the ImmutableArray you get back from an ArrayBuilder.